lms/templates/courseware/course_about.html

```diff
@@ -31,49 +36,54 @@
+            location.href = "${reg_then_add_to_cart_link| n, js_escaped_string}";
         }
       };
 
       $("#add_to_cart_post").click(function(event){
         $.ajax({
-          url: "${reverse('add_course_to_cart', args=[course.id.to_deprecated_string()])}",
+          url: "${reverse('add_course_to_cart', args=[course.id.to_deprecated_string()])| n, js_escaped_string}",
           type: "POST",
           /* Rant: HAD TO USE COMPLETE B/C PROMISE.DONE FOR SOME REASON DOES NOT WORK ON THIS PAGE. */
           complete: add_course_complete_handler
         })
         event.preventDefault();
       });
     % endif
-
     ## making the conditional around this entire JS block for sanity
     %if settings.FEATURES.get('RESTRICT_ENROLL_BY_REG_METHOD') and course.enrollment_domain:
       <%
-        perms_error = _('The currently logged-in user account does not have permission to enroll in this course. '
-                        'You may need to {start_logout_tag}log out{end_tag} then try the enroll button again. '
-                        'Please visit the {start_help_tag}help page{end_tag} for a possible solution.').format(
-                          start_help_tag="<a href='{url}'>".format(url=marketing_link('FAQ')), end_tag='</a>',
-                          start_logout_tag="<a href='{url}'>".format(url=reverse('logout'))
-                          )
+        perms_error = Text(
+                _('The currently logged-in user account does not have permission to enroll in this course. '
+                  'You may need to {start_logout_tag}log out{end_tag} then try the enroll button again. '
+                  'Please visit the {start_help_tag}help page{end_tag} for a possible solution.')
+                ).format(
+                  start_help_tag=HTML("<a href='{url}'>").format(url=marketing_link('FAQ')),
+                  end_tag=HTML('</a>'),
+                  start_logout_tag=HTML("<a href='{url}'>").format(url=reverse('logout'))
+                )
       %>
     $('#class_enroll_form').on('ajax:complete', function(event, xhr) {
       if(xhr.status == 200) {
-        location.href = "${reverse('dashboard')}";
+          location.href = "${reverse('dashboard')| n, js_escaped_string}";
       } else if (xhr.status == 403) {
-        location.href = "${reverse('course-specific-register', args=[course.id.to_deprecated_string()])}?course_id=${course.id | u}&enrollment_action=enroll";
-      } else if (xhr.status == 400) { //This means the user did not have permission
-        $('#register_error').html("${perms_error}").css("display", "block");
+          location.href = "${reverse('course-specific-register', args=[course.id.to_deprecated_string()])| n, js_escaped_string}?course_id=${course.id| n, js_escaped_string}&enrollment_action=enroll";
+      } else if (xhr.status == 400) {//This means the user did not have permission
+          edx.HtmlUtils.setHtml(
+                  $('#register_error'),
+                  "${perms_error}"
```

Comment:
@Qubad786: Hopefully this makes it even more clear the difference between JavaScript escaping and HTML escaping.  Let me know if this helps or is just confusing.

Python code showing JavaScript escaping:
```
>>> from openedx.core.djangolib.js_utils import js_escaped_string
>>> print js_escaped_string("<strong>Message with HTML.</strong>")
\u003Cstrong\u003EMessage with HTML.\u003C/strong\u003E
```

JQuery code showing how a JavaScript-escaped string is not HTML-escaped:
```
$('body').append("\u003Cstrong\u003EMessage with HTML.\u003C/strong\u003E")

<body>
...
<strong>Message with HTML.</strong>
</body>
```